test/authNZ.mjs

```diff
@@ -0,0 +1,166 @@
+import * as awilix from "awilix";
+import { expect } from "chai";
+import { beforeEach } from "mocha";
+import sinon, { stub } from "sinon";
+
+import { run } from "../src/cli.mjs";
+import { setupTestContainer as setupContainer } from "../src/config/setup-test-container.mjs";
+import { authNZMiddleware, setAccountKey } from "../src/lib/auth/authNZ.mjs";
+import { AccountKey, SecretKey } from "../src/lib/file-util.mjs";
+import { InvalidCredsError } from "../src/lib/misc.mjs";
+import { f } from "./helpers.mjs";
+
+describe("authNZMiddleware", function () {
+  let container;
+  let fetch;
+  let logger;
+  const validAccessKeyFile =
+    '{"test-profile": { "accountKey": "valid-account-key", "refreshToken": "valid-refresh-token"}}';
+  const validSecretKeyFile =
+    '{"valid-account-key": { "test-db": {"admin": "valid-db-key"}}}';
+  const mockAccountClient = () => {
+    return {
+      whoAmI: stub().resolves(true),
+      createKey: stub().resolves({ secret: "new-db-key" }),
+      refreshSession: stub().resolves({
+        account_key: "new-account-key",
+        refresh_token: "new-refresh-token",
+      }),
+    };
+  };
+
+  beforeEach(() => {
+    container = setupContainer();
+    container.register({
+      accountClient: awilix.asFunction(mockAccountClient).scoped(),
+      accountCreds: awilix.asClass(AccountKey).scoped(),
+      secretCreds: awilix.asClass(SecretKey).scoped(),
+    });
+    fetch = container.resolve("fetch");
+    logger = container.resolve("logger");
+  });
+
+  it("should pass through if authRequired is false", async function () {
+    const argv = { authRequired: false };
+    const result = await authNZMiddleware(argv);
+    expect(fetch.called).to.be.false;
+    expect(result).to.deep.equal(argv);
+  });
+
+  it("should prompt login if InvalidCredsError is thrown", async function () {
+    const scope = container.createScope();
+    const argv = { authRequired: true, profile: "test-profile" };
+    await run("db list", scope);
+    const exit = scope.resolve("exit");
+    const accountCreds = scope.resolve("accountCreds");
+
+    accountCreds.get = stub().throws(new InvalidCredsError());
+
+    await authNZMiddleware(argv);
+    expect(logger.stderr.args[0][0]).to.include("not signed in or has expired");
+    expect(logger.stdout.args[0][0]).to.include(
+      "To sign in, run:\n\nfauna login --profile test-profile\n",
+    );
```

Comment:
you can get this effect from this (more readable) syntax. sinon.match will do a substring search.

```
expect(logger.stderr).to.have.been.calledWith(sinon.match("not signed in or has expired"))
```

however, if you know what all of the output should look like, i added `scope.resolve("stdoutStream")` recently, which you can use like [this](https://github.com/fauna/fauna-shell/blob/v3/test/shell.mjs#L123) to make assertions against the whole output.